### PR TITLE
Add an algorithm to find mean absolute deviation

### PR DESCRIPTION
**Pull request by prashalruchiranga**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Maths/MeanAbsoluteDeviation.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+function meanAbsoluteDeviation (data) {
+  if (!Array.isArray(data)) {
+    throw new TypeError('Invalid Input')
+  }
+  let absoluteSum = 0
+  const meanValue = mean(data)
+  for (const dataPoint of data) {
+    absoluteSum += Math.abs(dataPoint - meanValue)
+  }
+  return parseFloat((absoluteSum / data.length).toFixed(3))
```
Comment (appgurueu) [inline Maths/MeanAbsoluteDeviation.js:18]: Why the `parseFloat`, and why truncate to the first 3 digits?

Comment (prashalruchiranga) [reply Maths/MeanAbsoluteDeviation.js:18]: I thought that much accuracy is not required. Therefore truncated to the first 3 decimals.

Comment (appgurueu) [reply Maths/MeanAbsoluteDeviation.js:18]: But how does truncating help with that? Truncation may be appropriate for tests, but not for the implementation / algorithm.

Comment (prashalruchiranga) [reply Maths/MeanAbsoluteDeviation.js]: Right. I have updated them.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**